### ISSUE TITLE
[BUG] Thetis will not communicate with CH340 COM:s

### ISSUE THREAD

**Issue by SM3PZG** [sjk7/Thetis-2.9.0#54]
**Describe the bug**
CAT2 and CAT3 control lost. 

 I use 2 CH340 USB to SERIAL adapters to control amplifier and antenna switching. Has been working great for 6 months. A few days ago starting Thetis 2.9.0.19 I got error enabling CAT2 on COM4. 

CAT1 is COM4 (com0com 4-5) and controls logging program and that still works fine. But the error message is COM4 when I activated COM9 on CAT2 or COM19 on CAT3. Error message should be COM9 and COM19 if those ports were not working, not COM4.

Tried different USB ports, uninstalled and reinstalled CH340 driver. Restart PC to get new COM numbers without luck. I then created 2 new virtual COM9 and COM19, restarted PC and connected CH340 adapter. I then got new COM6 and COM7.  Started Thetis and COM6 and COM7 worked fine on CAT2 and CAT3. 

So to my disappointment this morning when starting Thetis I get the same error message. "Error enabling CAT2 on COM4." COM4 is still on CAT1 and works fine with logging program. If COM6 or COM7 were not working error message should be Error enabling COM6 or COM7 on CAT2 or CAT3.  

Installed Thetis 2.9.0.29 this morning. Same problem with CH340 ports. If I try Enable other COM:s on CAT2 and CAT3 that works without error so its something wrong with Thetis communicating with CH340 ports. Or the other way around.

**To Reproduce**
Really don't know.

**Expected behaviour**
COM:s should work as they done for the last 6 months.

**Screenshots**

![CAT2 error](https://user-images.githubusercontent.com/123744177/236151820-d38e6221-eb4e-414b-a8d6-acd92bfcb128.png)
![CAT Thetis](https://user-images.githubusercontent.com/123744177/236151887-01d72918-657c-47cc-a503-9af02f2a2342.png)
![Task](https://user-images.githubusercontent.com/123744177/236151921-75735b7f-75da-479c-95e5-faf46b29c9b0.png)

**Desktop (please complete the following information):**
Windows 11 (last updated a month ago.) Andromeda Thetis 2.9.0.29 FW v2.1.26 Protocol 2

**Additional context**
Thanks Steve for your great work. Its pure joy every day starting up Thetis.

**Comment by RdWing** [comment]
Just a tip for CAT to other programs on the computer, generally you can do it over TCP by leveraging the CAT over TCPIP server built into thetis. On the client program looking for a connection, fill in the COM whatever box with the loopback ip and port, then you can advoid com0com or other serial port emulators all together.

**Comment by SM3PZG** [comment]
Sure. The CH340 drivers can be found here [http://www.wch-ic.com/downloads/CH341SER_EXE.html](url)
I downloaded CH341SER and removed all CH340 drivers. Then I installed them again with CH341SER. I started at terminal program to make sure COM:s were communicating witch they were. Started Thetis and still the same problem.  

All the com0com virtual com port pairs work fine [https://com0com.sourceforge.net/](url) Its the 2 CH340 USB to RS232 USB adapters that are causing some kind of conflict in Thetis.

**Comment by SM3PZG** [comment]
After several hours of installing/uninstalling drivers, changing port numbers and so on I got the 2 CH340 USB to RS232 working again. In Device manager I choose Show hidden units. There were several hidden COM:s and USB control units that I uninstalled. Rebooted. In Device manager I changed CH340 COM6 and COM7 to COM8 and COM9 and installed new drivers with CH341SER.

If a COM port is not working on CAT2 maybe Thetis could say for instance - Error enabling CAT2 on COM6? Not say the CAT1 COM port witch makes it confusing.

73/Sam
